pandas/core/series.py

```diff
@@ -2518,13 +2519,13 @@ def _binop(self, other, func, level=None, fill_value=None):
 
         with np.errstate(all='ignore'):
             result = func(this_vals, other_vals)
+
         name = ops.get_op_result_name(self, other)
-        result = self._constructor(result, index=new_index, name=name)
-        result = result.__finalize__(self)
-        if name is None:
-            # When name is None, __finalize__ overwrites current name
-            result.name = None
-        return result
+        if func.__name__ in ['divmod', 'rdivmod']:
+            ret = ops._construct_divmod_result(self, result, new_index, name)
```

Comment:
you could just return on each statement

Reply:
Previous review comment suggested me to use `_construct_divmod_result`.